### PR TITLE
Fix warnings: 1507 , 1707 , 1721

### PR DESCRIPTION
**Pull request by azabbasi**
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1507
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1707
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1721

### CODE REVIEW

**Review by barustum — APPROVED** [review]

File: iothub/device/src/Message.cs [hunk 1]
```diff
@@ -271,10 +255,15 @@ public string ComponentName
 
         bool IReadOnlyIndicator.IsReadOnly => Interlocked.Read(ref _sizeInBytesCalled) == 1;
 
+#pragma warning disable CA1721 // Property names should not match get methods; Justification: Cannot remove public property on a public facing type
```
Comment (abhipsaMisra) [inline iothub/device/src/Message.cs:258]: nit: suppress via attribute instead?

File: iothub/device/src/Message.cs [hunk 2]
```diff
@@ -113,22 +113,6 @@ public string CorrelationId
             set => SystemProperties[MessageSystemPropertyNames.CorrelationId] = value;
         }
 
-        /// <summary>
-        /// Indicates whether consumption or expiration of the message should post data to the feedback queue
-        /// </summary>
-        private DeliveryAcknowledgement Ack
```
Comment (abhipsaMisra) [inline iothub/device/src/Message.cs:119]: was this unreferenced?

File: iothub/device/src/Message.cs [hunk 3]
```diff
@@ -388,11 +372,6 @@ public void SetAsSecurityMessage()
             SystemProperties[MessageSystemPropertyNames.InterfaceId] = CommonConstants.SecurityMessageInterfaceId;
         }
 
-        private void SetSizeInBytesCalled()
```
Comment (abhipsaMisra) [inline iothub/device/src/Message.cs:391]: unreferenced?

File: iothub/device/src/Common/InternalBufferManager.cs [hunk 4]
```diff
@@ -330,7 +320,7 @@ public override void ReturnBuffer(byte[] buffer)
                 {
                     if (buffer.Length != bufferPool.BufferSize)
                     {
-                        throw Fx.Exception.Argument("buffer", CommonResources.BufferIsNotRightSizeForBufferManager);
+                        throw Fx.Exception.Argument(nameof(buffer), Common.Resources.BufferIsNotRightSizeForBufferManager);
```
Comment (abhipsaMisra) [inline iothub/device/src/Common/InternalBufferManager.cs:323]: question - how did this change from `CommonResources` to `Common.Resources`?

File: iothub/device/src/TransportType.cs [hunk 5]
```diff
@@ -6,11 +6,15 @@ namespace Microsoft.Azure.Devices.Client
...
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Naming",
+        "CA1707:Identifiers should not contain underscores",
+        Justification = "Public facing types cannot be renamed. This is considered a breaking change")]
     public enum TransportType
     {
         /// <summary>
         /// Advanced Message Queuing Protocol transport.
-        /// Try Amqp over TCP first and fallback to Amqp over WebSocket if that fails
+        /// Try AMQP over TCP first and fall-back to AMQP over WebSocket if that fails
```
Comment (drwill-ms) [inline iothub/device/src/TransportType.cs:17]: fall back is used as a verb, so no hyphen should be used

**Review by drwill-ms — APPROVED** [review]

File: iothub/device/src/Message.cs (hunk 2 above)
Comment (azabbasi) [reply iothub/device/src/Message.cs:119]: correct

File: iothub/device/src/Message.cs (hunk 3 above)
Comment (azabbasi) [reply iothub/device/src/Message.cs:391]: correct

File: iothub/device/src/Common/InternalBufferManager.cs (hunk 4 above)
Comment (azabbasi) [reply iothub/device/src/Common/InternalBufferManager.cs:323]: Common.Resources has the property that is implemented by CommonResources ... unnecessary change, but suggested by the IDE

File: iothub/device/src/TransportType.cs [hunk 6]
```diff
@@ -6,11 +6,15 @@ namespace Microsoft.Azure.Devices.Client
...
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Naming",
+        "CA1707:Identifiers should not contain underscores",
+        Justification = "Public facing types cannot be renamed. This is considered a breaking change")]
     public enum TransportType
     {
         /// <summary>
         /// Advanced Message Queuing Protocol transport.
-        /// Try Amqp over TCP first and fallback to Amqp over WebSocket if that fails
+        /// Try AMQP over TCP first and fallback to AMQP over WebSocket if that fails
```
Comment (drwill-ms) [inline iothub/device/src/TransportType.cs:17]: No hyphen but it is still two words: `fall back`

Comment (azabbasi) [reply iothub/device/src/TransportType.cs:17]: I will fix it in my incoming PR, don't want to re-run the checks again if there is nothing else.

**Pull request merged**